src/libtexture/imagecache.cpp

```diff
@@ -2822,6 +2822,48 @@ ImageCacheImpl::get_image_info(ImageCacheFile* file,
 #undef ATTR_DECODE
 }
 
+bool
+ImageCacheImpl::get_image_info_type(ustring filename, int subimage,
+                                    int miplevel, ustring dataname,
+                                    TypeDesc& datatype)
+{
+    ImageCachePerThreadInfo* thread_info = get_perthread_info();
+    ImageCacheFile* file = find_file(filename, thread_info, nullptr);
+    if (!file && dataname != s_exists) {
+        error("Invalid image file \"{}\"", filename);
+        return false;
+    }
+    return get_image_info_type(file, thread_info, subimage, miplevel, dataname,
+                               datatype);
+}
+
+bool
+ImageCacheImpl::get_image_info_type(ImageCacheFile* file,
+                                    ImageCachePerThreadInfo* thread_info,
+                                    int subimage, int miplevel,
+                                    ustring dataname, TypeDesc& datatype)
+{
+    // Output the TypeDesc of a given attribute (if found). If not found
+    // we set it to UNKNOWN.
+
+    ImageCacheStatistics& stats(thread_info->m_stats);
```

Comment:
See lines 2581ff, where get_image_info checks and gives appropriate errors and early exit for a null `file`, and lines 2672ff, where it handles non-existant subimage or miplevel. I think we need to do this here as well.

Reply:
Hmm... that comment doesn't seem to have gotten associated with the right spot in your code. :-)
I was trying to point this out in get_image_info_type, line 2853 (of your changes), right before you dereference `file` and ask for the spec for this specific subimage and miplevel.